workflows/src/test/kotlin/com/r3/corda/lib/tokens/workflows/TokenTypeBuilderTest.java

```diff
@@ -0,0 +1,30 @@
+package com.r3.corda.lib.tokens.workflows;
+
+import com.r3.corda.lib.tokens.money.DigitalCurrency;
+import com.r3.corda.lib.tokens.workflows.utilities.TokenBuilder;
+import net.corda.core.identity.CordaX500Name;
+import net.corda.core.identity.Party;
+import net.corda.testing.node.StartedMockNode;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class CurrencyAccessFromJavaTest extends JITMockNetworkTests {
+
+    /**
+     * Sanity check for easy access of digital currency TokenTypes from java code
+     */
+    @Test
+    public void javaWrappedDigitalCurrencyIsIdenticalToKotlinCompanionObject() throws Exception {
+        CordaX500Name aliceX500Name = new CordaX500Name("Alice", "NY", "US");
+        StartedMockNode alice = node(aliceX500Name);
+        Party aliceParty = alice.getInfo().getLegalIdentities().get(0);
+
+        new TokenBuilder()
```

Comment:
How do you get a token from the builder?

Reply:
@rogersanick this ^^ was also my next question.

With the kot infix the primary return vals will be these cases:
        val token : TokenType = GBP
        val amountTokenType : Amount<TokenType> = 10.GBP
        val nonFunIssued : IssuedTokenType = GBP issuedBy party
        val funIssued : Amount<IssuedTokenType> = 10.GBP issuedBy party
        val nonFunIssuedAndHeld : NonFungibleToken = GBP issuedBy party heldBy party
        val funIssuedAndHeld : FungibleToken = 10.GBP issuedBy party heldBy party

With the general constraints
amount (optional) -> tokenType [R]-> issuedBy [R] -> heldBy [R]

Not sure if this is the BEST practice, but could have a closing .resolve() which could be inserted in the [R] points to return the given object.

Reply:
... my bad, this one should have been obvious. Hurray for POC's!